### PR TITLE
Add micro editor

### PR DESCRIPTION
**Pull request by dhirschfeld**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [ ] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
> *This package needs to use a `git_url` as the build system relies on git to correctly version the package without which certain functionality including plugins won't work*
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by dhirschfeld** [comment]
Error on osx so will skip for now:
```
/Users/runner/miniforge3/conda-bld/micro_1618316836012/_build_env/go/pkg/tool/darwin_amd64/link: running /dev/null failed: fork/exec /dev/null: permission denied

```

**Comment by dhirschfeld** [comment]
```
go-licenses: command not found
```
🤦

**Comment by dhirschfeld** [comment]
Not sure what's going on with Windows
```
  File "C:\Miniconda\lib\site-packages\conda_build\source.py", line 251, in git_mirror_checkout_recursive
    assert isdir(mirror_dir)
AssertionError
```

File: recipes/micro/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+test:
+  commands:
+    - micro --version
+
+about:
+  home: https://micro-editor.github.io/
+  license: MIT
+  license_file:
+    - LICENSE
+    - "{{ RECIPE_DIR }}/thirdparty-licenses/"
```
Comment (dhirschfeld) [inline recipes/micro/meta.yaml:33]: Haven't been able to get this functionality to work - specifying a folder here always results in copy errors:
```
Traceback (most recent call last):
  File "/opt/conda/lib/python3.8/site-packages/conda_build/utils.py", line 537, in _copy_with_shell_fallback
    subprocess.check_call('cp -a {} {}'.format(src, dst), shell=True,
  File "/opt/conda/lib/python3.8/subprocess.py", line 364, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command 'cp -a /home/conda/staged-recipes-copy/recipes/micro/thirdparty-licenses/layeh.com/gopher-luar/array_test.go /home/conda/staged-recipes/build_artifacts/micro_1618367372622/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_place/info/licenses/LICENSE0/layeh.com/gopher-luar/array_test.go' returned non-zero exit status 1.
```

**Comment by dhirschfeld** [comment]
This recipe packages the licenses into the `/info/recipe` folder since I couldn't get the `conda-build` copying to work :/
<details>
<summary>Package Structure</summary>

```
micro-2.0.8-ha8f183a_0
│
├───bin
│       micro
│
└───info
    │   about.json
    │   files
    │   git
    │   hash_input.json
    │   index.json
    │   paths.json
    │
    ├───licenses
    │       LICENSE
    │
    ├───recipe
    │   │   bld.bat
    │   │   build.sh
    │   │   conda_build_config.yaml
    │   │   meta.yaml
    │   │   meta.yaml.template
    │   │
    │   └───thirdparty-licenses
    │       ├───github.com
    │       │   ├───blang
    │       │   │   └───semver
    │       │   │           LICENSE
    │       │   │
    │       │   ├───dustin
    │       │   │   └───go-humanize
    │       │   │           LICENSE
    │       │   │
    │       │   ├───gdamore
    │       │   │   └───encoding
    │       │   │           LICENSE
    │       │   │
    │       │   ├───go-errors
    │       │   │   └───errors
    │       │   │           LICENSE.MIT
    │       │   │
    │       │   ├───kballard
    │       │   │   └───go-shellquote
    │       │   │           LICENSE
    │       │   │
    │       │   ├───lucasb-eyer
    │       │   │   └───go-colorful
    │       │   │           LICENSE
    │       │   │
    │       │   ├───mattn
    │       │   │   ├───go-isatty
    │       │   │   │       LICENSE
    │       │   │   │
    │       │   │   └───go-runewidth
    │       │   │           LICENSE
    │       │   │
    │       │   ├───mitchellh
    │       │   │   └───go-homedir
    │       │   │           LICENSE
    │       │   │
    │       │   ├───sergi
    │       │   │   └───go-diff
    │       │   │       └───diffmatchpatch
    │       │   │               LICENSE
    │       │   │
    │       │   ├───xo
    │       │   │   └───terminfo
    │       │   │           LICENSE
    │       │   │
    │       │   ├───yuin
    │       │   │   └───gopher-lua
    │       │   │           LICENSE
    │       │   │
    │       │   └───zyedidia
    │       │       ├───clipboard
    │       │       │       LICENSE
    │       │       │
    │       │       ├───glob
    │       │       │       LICENSE
    │       │       │
    │       │       ├───json5
    │       │       │       LICENSE
    │       │       │
    │       │       ├───micro
    │       │       │   └───v2
    │       │       │           LICENSE
    │       │       │
    │       │       ├───pty
    │       │       │       License
    │       │       │
    │       │       ├───tcell
    │       │       │   └───v2
    │       │       │           LICENSE
    │       │       │
    │       │       └───terminal
    │       │               LICENSE
    │       │
    │       ├───golang.org
    │       │   └───x
    │       │       ├───sys
    │       │       │   └───unix
    │       │       │           LICENSE
    │       │       │
    │       │       └───text
    │       │               LICENSE
    │       │
    │       ├───gopkg.in
    │       │   └───yaml.v2
    │       │           LICENSE
    │       │           NOTICE
    │       │
    │       └───layeh.com
    │           └───gopher-luar
    │                   .gitignore
    │                   array.go
    │                   array_test.go
    │                   cache.go
    │                   chan.go
    │                   chan_test.go
    │                   config.go
    │                   config_test.go
    │                   doc.go
    │                   examples_test.go
    │                   example_test.go
    │                   func.go
    │                   func_test.go
    │                   go.mod
    │                   go.sum
    │                   helpers_test.go
    │                   LICENSE
    │                   luar.go
    │                   luar_test.go
    │                   Makefile
    │                   map.go
    │                   map_iter.go
    │                   map_iter_go112.go
    │                   map_test.go
    │                   metatable.go
    │                   metatable_test.go
    │                   ptr.go
    │                   ptr_test.go
    │                   race_test.go
    │                   README.md
    │                   slice.go
    │                   slice_test.go
    │                   struct.go
    │                   struct_test.go
    │                   type.go
    │                   type_test.go
    │                   util.go
    │
    └───test
            run_test.sh
```
</details>

Perhaps the copy errors are due to the go files being copied into the licenses folder?

**Comment by dhirschfeld** [comment]
ping @conda-forge/staged-recipes this is ready for review.

File: recipes/micro/build.sh [hunk 2]
```diff
@@ -0,0 +1,9 @@
+#! /usr/bin/env bash
+
+set -eoux pipefail
+
+
+make build
+make install
```
Comment (ocefpaf) [inline recipes/micro/build.sh:7]: Is there a make check/test?

Comment (dhirschfeld) [reply recipes/micro/build.sh:7]: It seems so!
https://github.com/zyedidia/micro/blob/f1aaa307436f0ee3fbf0acc5b490cf328662de82/Makefile#L66-L68

**Comment by dhirschfeld** [comment]
Tests seem to pass:
```
go test ./internal/...
go: downloading github.com/stretchr/testify v1.4.0
go: downloading github.com/pmezard/go-difflib v1.0.0
go: downloading github.com/davecgh/go-spew v1.1.1
?   	github.com/zyedidia/micro/v2/internal/action	[no test files]
ok  	github.com/zyedidia/micro/v2/internal/buffer	0.041s
?   	github.com/zyedidia/micro/v2/internal/clipboard	[no test files]
ok  	github.com/zyedidia/micro/v2/internal/config	0.012s
?   	github.com/zyedidia/micro/v2/internal/display	[no test files]
?   	github.com/zyedidia/micro/v2/internal/info	[no test files]
?   	github.com/zyedidia/micro/v2/internal/lua	[no test files]
?   	github.com/zyedidia/micro/v2/internal/screen	[no test files]
?   	github.com/zyedidia/micro/v2/internal/shell	[no test files]
ok  	github.com/zyedidia/micro/v2/internal/util	0.003s
ok  	github.com/zyedidia/micro/v2/internal/views	0.011s
go test ./cmd/...
ok  	github.com/zyedidia/micro/v2/cmd/micro	0.098s
```

**Pull request merged**